can/interfaces/socketcan/socketcan.py

```diff
@@ -435,34 +433,34 @@ def __init__(self, channel, receive_own_messages=False, fd=False, **kwargs):
         :param list can_filters:
             See :meth:`can.BusABC.set_filters`.
         """
-        self.socket = create_socket(CAN_RAW)
+        self.socket = create_socket()
         self.channel = channel
         self.channel_info = "native socketcan channel '%s'" % channel
 
         # set the receive_own_messages paramater
         try:
-            self.socket.setsockopt(socket.SOL_CAN_RAW,
-                                   socket.CAN_RAW_RECV_OWN_MSGS,
+            self.socket.setsockopt(SOL_CAN_RAW,
+                                   CAN_RAW_RECV_OWN_MSGS,
                                    struct.pack('i', receive_own_messages))
         except socket.error as e:
             log.error("Could not receive own messages (%s)", e)
 
         if fd:
             # TODO handle errors
-            self.socket.setsockopt(socket.SOL_CAN_RAW,
-                                   socket.CAN_RAW_FD_FRAMES,
+            self.socket.setsockopt(SOL_CAN_RAW,
+                                   CAN_RAW_FD_FRAMES,
                                    struct.pack('i', 1))
 
         bind_socket(self.socket, channel)
 
         kwargs.update({'receive_own_messages': receive_own_messages, 'fd': fd})
-        super(SocketcanNative_Bus, self).__init__(channel=channel, **kwargs)
+        super(SocketcanBus, self).__init__(channel=channel, **kwargs)
 
     def shutdown(self):
         self.socket.close()
 
     def _recv_internal(self, timeout):
-        if timeout:
+        if timeout is not None:
```

Comment:
A timeout of `0` should not try to invoke `select()`, or should it?

Reply:
It should, because otherwise a timeout of 0 will go directly to `recv` which will block with infinite timeout.